plugins/moveToArchive.koplugin/main.lua

```diff
@@ -0,0 +1,134 @@
+local DataStorage = require("datastorage")
+local ReadHistory = require("readhistory")
+local FileManager = require("apps/filemanager/filemanager")
+local InfoMessage = require("ui/widget/infomessage")
+local LuaSettings = require("frontend/luasettings")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local util = require("frontend/util")
+local _ = require("gettext")
+
+local MoveToArchive = WidgetContainer:new{
+    name = "move2archive",
+}
+
+local move_to_archive_settings = LuaSettings:open(("%s/%s"):format(DataStorage:getSettingsDir(), "move_to_archive_settings.lua"))
+local archive_dir_config_key = "archive_dir";
+local last_copied_from_config_key = "last_copied_from_dir";
+local archive_dir_path = move_to_archive_settings:readSetting(archive_dir_config_key)
+local last_copied_from_dir = move_to_archive_settings:readSetting(last_copied_from_config_key)
+
+function MoveToArchive:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function MoveToArchive:addToMainMenu(menu_items)
+    menu_items.moveToArchive = {
+        text = _("Move to Archive"),
+        sub_item_table = {
+            {
+                text = _("Move current book to archive"),
+                callback = self.moveToArchive,
+            },
+            {
+                text = _("Copy current book to archive"),
+                callback = self.copyToArchive,
+            },
+            {
+                text = _("Go to archive folder"),
+                callback = function()
+                    if not archive_dir_path then
+                        UIManager:show(InfoMessage:new{
+                            text = _("No archive directory. Please set it first")
```

Comment:
You could make it a simple ConfirmBox.

Please set an archive folder first.

Cancel/Set archive folder